db/etcd.go

```diff
@@ -27,499 +27,336 @@ import (
 // in the database as needed.
-func (etcddb etcdDB) SetAgent(adv defs.AgentAdvert) {
-
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
-
+func (etcddb *etcdDB) SetAgent(adv defs.AgentAdvert) error {
 	log.Infof("Setting '/todd/agents/%s' key", adv.Uuid)
 
-	adv_json, err := json.Marshal(adv)
+	advJSON, err := json.Marshal(adv)
 	if err != nil {
 		log.Error("Problem converting Agent Advertisement to JSON")
-		os.Exit(1)
+		return err
 	}
 
 	// TODO(mierdin): TTL needs to be user-configurable
-	resp, err := kapi.Set(
+	resp, err := etcddb.keysAPI.Set(
 		context.Background(),                      // context
 		fmt.Sprintf("/todd/agents/%s", adv.Uuid),  // key
-		string(adv_json),                          // value
+		string(advJSON),                           // value
 		&client.SetOptions{TTL: time.Second * 30}, //optional args
 	)
 	if err != nil {
 		log.Error("Problem setting agent in etcd")
-		os.Exit(1)
+		return err
 	}
 
 	log.Infof("Agent set in etcd. This advertisement is good until %s", resp.Node.Expiration)
 
+	return nil
 }
 
 // GetAgent will retrieve a specific agent from the database by UUID
-func (etcddb etcdDB) GetAgent(uuid string) defs.AgentAdvert {
+func (etcddb *etcdDB) GetAgent(uuid string) (*defs.AgentAdvert, error) {
+	log.Printf("Getting /todd/agents/%s' key value", uuid)
 
-	adv := defs.AgentAdvert{}
+	keyStr := fmt.Sprintf("/todd/agents/%s", uuid)
 
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-	c, err := client.New(cfg)
+	resp, err := etcddb.keysAPI.Get(context.Background(), keyStr, &client.GetOptions{Recursive: true})
 	if err != nil {
-		log.Fatal(err)
+		log.Errorf("Agent %s not found.", uuid)
+		return nil, err
 	}
-	kapi := client.NewKeysAPI(c)
 
-	log.Printf("Getting /todd/agents/%s' key value", uuid)
-
-	key_str := fmt.Sprintf("/todd/agents/%s", uuid)
+	log.Debugf("Etcd 'get' is done. Metadata is %q\n", resp)
 
-	resp, err := kapi.Get(context.Background(), key_str, &client.GetOptions{Recursive: true})
+	adv, err := nodeToAgentAdvert(resp.Node, uuid)
 	if err != nil {
-		log.Errorf("Agent %s not found.", uuid)
-		return adv
+		return nil, err
 	}
 
-	log.Debugf("Etcd 'get' is done. Metadata is %q\n", resp)
+	return adv, nil
+}
+
+func nodeToAgentAdvert(node *client.Node, expectedUUID string) (*defs.AgentAdvert, error) {
```

Comment:
Yup! Just going after consistency, primarily. I like all functions to have at least a little context describing their purpose